osquery/tables/utility/file.cpp

```diff
@@ -25,8 +32,144 @@ namespace osquery {
 
 namespace tables {
 
-#if !defined(WIN32)
+namespace {
+#ifdef WIN32
+
+struct LnkData {
+  std::string target_path;
+  std::string target_type;
+  std::string target_location;
+  std::string start_in;
+  std::string run;
+  std::string comment;
+};
+
+std::string showCmdToString(int show_cmd) {
+  switch (show_cmd) {
+  case SW_SHOWNORMAL: {
+    return "Normal window";
+  }
+  case SW_SHOWMAXIMIZED: {
+    return "Maximized";
+  }
+  case SW_SHOWMINIMIZED: {
+    return "Minimized";
+  }
+  default: {
+    return "Unknown";
+  }
+  }
+}
+
+boost::optional<LnkData> parseLnkData(const fs::path& link) {
+  IShellLink* shell_link;
```

Comment:
Does `shell_link` need to be freed as well? Judging by the example in https://learn.microsoft.com/en-us/windows/win32/shell/links#resolving-a-shortcut it seems like it may need to be.